Documentation/security/network/encryption-wireguard.rst

```diff
@@ -295,6 +295,45 @@ options:
   - LoadBalancer & NodePort XDP Acceleration
   - Direct Server Return (DSR)
```

Comment:
We should also mention the EgressGW reply path here (with XDP).

Can you elaborate on why DSR doesn't get encrypted? :thinking:

Reply:
> We should also mention the EgressGW reply path here (with XDP).

Aha :+1: Do we have a GH issue for it?

> Can you elaborate on why DSR doesn't get encrypted? 🤔

The src IP of the forwarded DSR packet is the one from the original client (outside cluster). Such a request cannot be forwarded by WG, as it drops packets if the IP does not belong to WG's allowed peer IPs.

Reply:
> > We should also mention the EgressGW reply path here (with XDP).
> 
> Aha 👍 Do we have a GH issue for it?
> 

Nope. This would basically be a "wireguard doesn't support XDP-internal forwarding", correct?

> > Can you elaborate on why DSR doesn't get encrypted? 🤔
> 
> The src IP of the forwarded DSR packet is the one from the original client (outside cluster). Such a request cannot be forwarded by WG, as it drops packets if the IP does not belong to WG's allowed peer IPs.

:ty:. So Geneve-DSR would actually be ok wrt encryption :).